src/view/dataElements/xdmObject/helpers/object/validate.js

```diff
@@ -33,20 +34,62 @@ export default ({
   }
 
   if (properties) {
+    const namesOfPopulatedProperties = new Set();
     const propertyNames = Object.keys(properties);
-    const propertyErrors = propertyNames.reduce((memo, propertyName) => {
-      const propertyFormStateNode = properties[propertyName];
-      const error = validate({
-        formStateNode: propertyFormStateNode,
-        isParentAnArray: false,
-        notifyParentOfDataPopulation: confirmDataPopulatedAtCurrentOrDescendantNode
-      });
-
-      if (error) {
-        memo[propertyName] = error;
-      }
-      return memo;
-    }, {});
+    let propertyErrors = propertyNames.reduce(
+      (propertyErrorsMemo, propertyName) => {
+        const propertyFormStateNode = properties[propertyName];
+        const error = validate({
+          formStateNode: propertyFormStateNode,
+          confirmDataPopulatedAtCurrentOrDescendantNode() {
+            namesOfPopulatedProperties.add(propertyName);
+            confirmDataPopulatedAtCurrentOrDescendantNode();
+          }
+        });
+
+        if (error) {
+          propertyErrorsMemo[propertyName] = error;
+        }
+        return propertyErrorsMemo;
+      },
+      {}
+    );
+
+    // Properties marked required are only actually required if the property's owning
+    // object exists in the XDM payload. The owning object will only be included in
+    // in the XDM payload if the object has at least one property with a populated value.
+    // At this point in the code, we know the names of the object properties that
+    // have a populated value. If at least one property has a populated value,
+    // we know the owning object will exist in the XDM payload, so we need to
+    // find any other properties in the object's schema that are required but
+    // don't yet have values. We'll ensure that these properties are given an error
+    // stating that the property is required.
+    // https://jira.corp.adobe.com/browse/PDCL-4413
+    if (namesOfPopulatedProperties.size) {
+      propertyErrors = propertyNames.reduce(
+        (propertyErrorsMemo, propertyName) => {
+          if (
+            // If the property is already populated, it won't qualify for an error.
+            !namesOfPopulatedProperties.has(propertyName) &&
+            // If the property already has some other type of error, we won't
+            // override it with a "required" error.
+            !propertyErrorsMemo[propertyName]
+          ) {
+            const propertyFormStateNode = properties[propertyName];
+            if (
+              propertyFormStateNode.schema.isRequired &&
+              propertyFormStateNode.autoPopulationSource === NONE
+            ) {
+              propertyErrorsMemo[propertyName] = {
+                value: "This is a required field and must be populated."
+              };
+            }
+          }
+          return propertyErrorsMemo;
+        },
+        propertyErrors
+      );
+    }
```

Comment:
This is the code that fixes the issue.

Reply:
As a side note, I'm much of a fan of `reduce` for the two cases above as I feel like it makes the code unnecessarily complex and difficult to follow. I did it anyway because that seems to follow our [convention](https://github.com/adobe/alloy/wiki/Conventions#reduce-instead-of-foreach).